engine/src/main/java/org/terasology/rendering/world/WorldRendererImpl.java

```diff
@@ -374,19 +351,7 @@ public void render(RenderingStage renderingStage) {
     private void renderShadowMap() {
         if (renderingConfig.isDynamicShadows() && isFirstRenderingStageForCurrentFrame) {
             PerformanceMonitor.startActivity("Render World (Shadow Map)");
-
-            graphicState.preRenderSetupSceneShadowMap();
-            shadowMapCamera.lookThrough();
-
-            renderChunks(renderQueues.chunksOpaqueShadow, ChunkMesh.RenderPhase.OPAQUE, shadowMapCamera, ChunkRenderMode.SHADOW_MAP);
-
-            for (RenderSystem renderer : systemManager.iterateRenderSubscribers()) {
-                renderer.renderShadows();
-            }
-
-            playerCamera.lookThrough(); // not strictly needed: just defensive programming here.
-            graphicState.postRenderCleanupSceneShadowMap();
-
+            shadowMap.render();
```

Comment:
I am thinking it might not be appropriate to call this method .render(). The graph will eventually contain nodes that are not render nodes, i.e. cameras and acceleration structures. For those it wouldn't make sense to have a "render()" method. And the graph should be transversed by calling the same method on each node as the graph is not really supposed to know much about what nodes do. So, I'd recommend something like "node.execute()" or something similar.